### PR TITLE
[WASI-NN] Unified preload options & support streaming to stdout

### PR DESCRIPTION
**Pull request by dm4**
- [WASI-NN] Unified preload options with case-insensitive matching 
- [WASI-NN] Add AUTO device support in ggml backend 
- [WASI-NN] Add STREAM_TO_STDOUT support in ggml backend

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: plugins/wasi_nn/wasinnenv.cpp [hunk 1]
```diff
@@ -17,16 +17,16 @@ create(const Plugin::PluginModule::ModuleDescriptor *) noexcept {
...
+    {"onnx"sv, Backend::ONNX},
+    {"tensorflow"sv, Backend::Tensorflow},
+    {"pytorch"sv, Backend::PyTorch},
+    {"tensorflowlite"sv, Backend::TensorflowLite},
+    {"autodetect"sv, Backend::Autodetect},
+    {"ggml"sv, Backend::GGML}};
 
 std::map<std::string_view, Device> DeviceMap = {
-    {"CPU"sv, Device::CPU}, {"GPU"sv, Device::GPU}, {"TPU"sv, Device::TPU}};
+    {"cpu"sv, Device::CPU}, {"gpu"sv, Device::GPU}, {"tpu"sv, Device::TPU}};
```
Comment (juntao) [inline plugins/wasi_nn/wasinnenv.cpp:29]: Can we add AUTO here? Thanks.

Comment (dm4) [reply plugins/wasi_nn/wasinnenv.cpp]: Added.

**Comment by dm4** [comment]
Will sperate `STREAM_TO_STDOUT` to another PR.

**Review by hydai — APPROVED** [review]

**Comment by dm4** [comment]
Nix build on CI:
> The hosted runner: GitHub Actions 148 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

**Comment by hydai** [comment]
> Nix build on CI:
> 
> > The hosted runner: GitHub Actions 148 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.
> 
> 

Ignore Nix CI temporarily

**Review by hydai — APPROVED** [review]

**Review by hydai — APPROVED** [review]

**Pull request merged**